internal/pkg/mapper/configmetadata/remote.go

```diff
@@ -0,0 +1,86 @@
+package configmetadata
+
+import (
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/remote"
+	"github.com/keboola/keboola-as-code/internal/pkg/state"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+)
+
+// configMetadataMapper add metadata to configurations loaded from API.
+type configMetadataMapper struct {
+	dependencies
+	logger         log.Logger
+	configMetadata map[model.BranchId]remote.ConfigMetadataResponse
+}
+
+type dependencies interface {
+	StorageApi() (*remote.StorageApi, error)
+}
+
+func NewMapper(s *state.State, d dependencies) *configMetadataMapper {
+	return &configMetadataMapper{dependencies: d, logger: s.Logger(), configMetadata: make(map[model.BranchId]remote.ConfigMetadataResponse)}
+}
+
+func (m *configMetadataMapper) OnRemoteChange(changes *model.RemoteChanges) error {
+	// Process loaded objects
+	errors := utils.NewMultiError()
+	for _, objectState := range changes.Loaded() {
+		if err := m.onRemoteLoad(objectState); err != nil {
+			errors.Append(err)
+		}
+	}
+
+	if errors.Len() > 0 {
+		// Convert errors to warning
+		m.logger.Warn(utils.PrefixError(`Warning`, errors))
+	}
+	return nil
+}
+
+func (m *configMetadataMapper) onRemoteLoad(objectState model.ObjectState) error {
+	config, ok := objectState.RemoteState().(*model.Config)
+	if !ok {
+		return nil
+	}
+	metadata, found, err := m.GetMetadata(config.ConfigKey)
+	if err != nil {
+		return err
+	}
+	if !found {
+		return nil
+	}
+	config.Metadata = metadata
+	return nil
+}
+
+func (m *configMetadataMapper) GetMetadata(configKey model.ConfigKey) (map[string]string, bool, error) {
+	configMetadataForBranch, ok := m.configMetadata[configKey.BranchId]
+	if !ok {
+		res, err := m.GetMetadataFromApi(configKey.BranchId)
+		if err != nil {
+			return nil, false, err
+		}
+		configMetadataForBranch = *res
+		m.configMetadata[configKey.BranchId] = configMetadataForBranch
+	}
```

Comment:
Pre kazdu branch robis blokujuci request, ... ak bude mat user 5x branch, tak to bude trvat 5x tak dlho.